Bricks/net/tcp/impl/posix.h

```diff
@@ -505,29 +501,44 @@ class Socket final : public SocketHandle {
   void operator=(Socket&&) = delete;
 };
 
+// Smart wrapper for system-allocated `addrinfo` pointers.
+using addrinfo_t = std::unique_ptr<struct addrinfo, std::function<void(struct addrinfo*)>>;
+
+inline addrinfo_t GetAddrInfo(const std::string& host, const std::string& serv = "") {
+  struct addrinfo* result = nullptr;
+  struct addrinfo hints;
+  memset(&hints, 0, sizeof(hints));
+  hints.ai_family = AF_INET;
+  hints.ai_socktype = SOCK_STREAM;
+  hints.ai_protocol = IPPROTO_TCP;
+  const int retval = ::getaddrinfo(host.c_str(), serv.c_str(), &hints, &result);
+  if (!result) {
+    CURRENT_THROW(SocketResolveAddressException());
+  } else if (retval) {
+    freeaddrinfo(result);
+    CURRENT_THROW(SocketResolveAddressException(gai_strerror(retval)));
+  }
+  return addrinfo_t(result, [](struct addrinfo* p) { ::freeaddrinfo(p); });
```

Comment:
Nit: The deleter is best to be put into `addrinfo_t`'s second template parameter.